### PR TITLE
Feature/tauri logger

### LINKED ISSUE #72: Implement backend logging to file as well as console
There are various logging crates out there that expand on the basic system `env_logger` uses to log into the console. Ideally, what we should have is logging to the file (to help with debugging) as well as some logging to console (with sane defaults for which logging levels are enabled by default).

### PR DESCRIPTION
**Pull request by medzernik**
This is a logging system using the Tauri backend. Might be better overall for the Macro App. Fixes #72

### CODE REVIEW

File: wooting-macro-backend/src/lib.rs [hunk 1]
```diff
@@ -552,8 +543,6 @@ impl MacroBackend {
                         rdev::EventType::ButtonRelease(button) => {
                             debug!("Button released: {:?}", button);
 
-                            buttons_pressed.blocking_write().retain(|x| *x != button);
-
```
Comment (simon-wh) [inline wooting-macro-backend/src/lib.rs:556]: ?

Comment (medzernik) [reply wooting-macro-backend/src/lib.rs:556]: the variable was never used for anything useful as you can only have one mouse button pressed at a time and that is already provided by the event. It was originally meant to be there to check multiple combinations of buttons or key presses before we decided to drop it. Thus, in one of the branches i noticed this not being needed and removed it but during the merge it got back in

File: wooting-macro-backend/src/lib.rs [hunk 2]
```diff
@@ -378,9 +376,15 @@ fn check_macro_execution_efficiently(
...
 
 impl MacroBackend {
-    #[cfg(not(debug_assertions))]
     /// Creates the data directory if not present in %appdata% (only in release build).
     pub fn generate_directories() {
+        #[cfg(not(debug_assertions))]
+        match std::fs::remove_file(config::LogFileName::file_name()) {
+            Ok(_) => info!("removed old log file"),
+            Err(error) => error!("Couldnt remove old log file!: {}", error),
+        }
```
Comment (simon-wh) [inline wooting-macro-backend/src/lib.rs:385]: ?

File: src/main.tsx [hunk 3]
```diff
@@ -5,6 +5,15 @@ import { theme } from './theme/index'
...
 import { SettingsProvider } from './contexts/settingsContext'
 import '@fontsource/montserrat/800.css'
+import { warn, info, error, attachConsole } from 'tauri-plugin-log';
+
+
+// with LogTarget::Webview enabled this function will print logs to the browser console
+const detach = await attachConsole();
+warn("Warn");
+info("Info");
+error("Error");
```
Comment (simon-wh) [inline src/main.tsx:15]: Remove

Comment (medzernik) [reply src/main.tsx]: Fixed in 286b9c8

File: wooting-macro-backend/src/lib.rs (hunk 2 above)
Comment (medzernik) [reply wooting-macro-backend/src/lib.rs]: Must have pulled in from the other branch, no need to remove the log.. Fixed in a02f397

File: src-tauri/src/main.rs [hunk 4]
```diff
@@ -77,14 +81,15 @@ async fn control_grabbing(
...
+        .and_then(|s| log::LevelFilter::from_str(s).ok())
+        .unwrap_or(log::LevelFilter::Info);
 
-    #[cfg(not(debug_assertions))]
     wooting_macro_backend::MacroBackend::generate_directories();
 
     let backend = MacroBackend::default();
 
-    info!("Running the macro backend");
+    println!("Running the macro backend");
```
Comment (simon-wh) [inline src-tauri/src/main.rs:92]: why `println`?

Comment (medzernik) [reply src-tauri/src/main.rs]: Fixed in 53b377e. I think i assumed that i can't use the logging until it's properly initialized and then forgot to change it back.

**Review by simon-wh — APPROVED** [review]

**Pull request merged**